### ISSUE TITLE
$regex operator mutates input query object

### ISSUE THREAD

**Issue by MattIPv4** [kofrasa/mingo#195]
👋 Very minor thing that we worked around by deep cloning our query, but it seems like the `$regex` operator results in the query object being mutated:

```js
const { Query } = require('mingo');
const assert = require('assert').strict;

// Create the object with the query
const queryObj = {
  type: { $regex: ".{2,}", $options: "i" },
};

// Confirm that deep equal sees it as equal
assert.deepEqual(queryObj, {
  type: { $regex: ".{2,}", $options: "i" },
});

// Peform the query, mutating the query object
new Query(queryObj).test({ type: "hello" });

// Check if the query object has been mutated
assert.deepEqual(queryObj, {
  type: { $regex: ".{2,}", $options: "i" },
});
```

(https://runkit.com/mattipv4/6151a7728cc9ee0008935c81)

**Comment by kofrasa** [comment]
Yeah, this is an old one. It is the result of normalizing the regex into a single value so that it can used as a single key/value in the operator map. You will notice that the example gets rewritten as `/.{2,}/i` which should be equivalent.

See https://github.com/kofrasa/mingo/blob/661faf187c61db483f183daf6aadec15c64c2a19/src/util.ts#L914-L915

It is an easy fix if you want to send a PR. We just have to return a new object from the normalized regex expression. 

If you do send a PR, please bump the version to `4.2.1` and update the `CHANGELOG` so I can just merge and publish.
